dashboard/app/controllers/courses_controller.rb

```diff
@@ -20,4 +29,21 @@ def show
 
     render 'show', locals: {course: course}
   end
+
+  def new
+    render 'new'
+  end
+
+  def create
+    course = Course.new(name: params.require(:course).require(:name))
+    if course.save
+      redirect_to action: :edit, course_name: course.name
+    else
+      render 'new', locals: {error_messages: course.errors.full_messages}
```

Comment:
Or if you prefer it to be explicit, pass `course` as a local, do `form_for course`, and inspect `course.errors`